### ISSUE TITLE
support mysql8 cluster

### ISSUE THREAD

**Issue by little-circle-circle** [liquibase/liquibase#3212]
hello, I use liquibase with mysql8 cluster,  start error, report DATABASECHANGELOG lack primary id. I check master branch, it still exist this problem

hope modify CreateDatabaseChangeLogTableGenerator like this
`.addColumn("ID", DataTypeFactory.getInstance().fromDescription(charTypeName + "(" + getIdColumnSize() + ")", database), null, null, new NotNullConstraint())`
->
`.addPrimaryKeyColumn("ID", DataTypeFactory.getInstance().fromDescription(charTypeName + "(" + getIdColumnSize() + ")", database), null, null, null, new NotNullConstraint())`

now, I extend CreateDatabaseChangeLogTableGenerator to do this, tnks!

**Comment by kataggart** [comment]
@little-circle-circle I am not following your issue. I think you are requesting we enhance the product to have a primary key in DATABASECHANGELOG - is that what you are requesting? Thanks.

**Comment by little-circle-circle** [comment]
@kataggart  yes, tables without a primary key can not be successfully created in MySQL Cluster, tnks.

**Comment by joelgriffiths** [comment]
I've experienced this issue far downstream as well. keycloak is using Liquidebase, and the missing primary key breaks kubernetes deployments into MySQL. This goes all the way back to https://liquibase.jira.com/browse/CORE-1909.

**Comment by filipelautert** [comment]
We're currently triaging our backlog to ensure we can focus on the issues that matter most. With a large number of open issues, it becomes difficult to give each one the attention it deserves.

This issue has been closed due to inactivity — there have been no updates for over 30 days. If this is still relevant to you, please feel free to reopen it or create a new issue with updated context. We appreciate your contribution!
